docs/source/tutorial_vectorarray.md

```diff
@@ -0,0 +1,687 @@
+by {{ VectorArrays }}, ordered collections of abstract vectors of the same
+dimension.
+The reason for this is that when pyMOR is used in conjunction with an external
+solver that implements the FOM, state space vectors must be compatible with the
+solver's internal data structures, for instance when some {{ Operator }}
+appearing in the FOM shall be applied to the vector.
+Instead of constantly converting between NumPy arrays and the solver's data structures
+-- which might even be impossible in some cases -- pyMOR uses {{ VectorArrays }} to access
+these vectors through a unified interface.
+So as soon as get in touch with a model's state-space data, e.g., by calling its
+{meth}`~pymor.models.interface.Model.solve` method, you will work with {{ VectorArrays }}.
+Note that this also the case for ROM. pyMOR does not distinguish between FOMs and ROMs,
+so even though no external solver is involved, the ROM state is still represented by
+{{ VectorArrays }}.
+
+## VectorSpaces / How to create VectorArrays
+
+{{ VectorArrays }} are never instantiated directly but are always created by the
+{{ VectorSpace }} they belong to.
+You will find {{ VectorSpaces }} as the {attr}`~pymor.operators.interface.Operator.source`
+and {attr}`~pymor.operators.interface.Operator.range` space of {{ Operators }} or as the
+{attr}`~pymor.models.interface.Model.solution_space` of {{ Models }}.
+In the following, we will work with {{ NumpyVectorArrays }} that internally store the
+vectors as a two-dimensional NumPy arrays.
+The corresponding {{ VectorSpace }} is called {{ NumpyVectorSpace }}.
+To make a {{ NumpyVectorSpace }}, we need to specify the dimension of the contained vectors:
+
+```{code-cell}
+from pymor.basic import NumpyVectorSpace
+import numpy as np
+
+space = NumpyVectorSpace(7)
+```
+
+The dimension can be accessed via the {attr}`~pymor.vectorarrays.interface.VectorSpace.dim`
+attribute:
+
+```{code-cell}
+space.dim
+```
+
+Our first {{ VectorArray }} will only contain three zero vectors:
+
+```{code-cell}
+U = space.zeros(3)
+```
+
+To check that we indeed have created three zero vectors of dimension 7, we can do the following:
+
+```{code-cell}
+print(len(U))
+print(U.dim)
+print(U.norm())
+```
+
+Just zero vectors are boring.
+To create vectors with different entries, we can use the following methods:
+
+```{code-cell}
+print(space.ones(1))
+print(space.full(42, count=4))
+print(space.random(2))
+```
+
+Sometimes, for instance when accumulating vectors in a loop, we want to initialize a
+{{ VectorArray }} with no vectors in it.
+We can use the {meth}`~pymor.vectorarrays.interface.VectorSpace.empty` method for that:
+
+```{code-cell}
+U.empty()
+```
+
+You might wonder how to create {{ VectorArrays }} with more interesting data.
+When implementing {{ Operators }} or, for instance the
+{meth}`~pymor.models.interface.Model.solve` method, you will get in touch with data
+from the actual solver backend you are using, that needs to be wrapped into a
```

Comment:
```suggestion
from the actual solver backend you are using that needs to be wrapped into a
```

Reply:
Disagree here. That's a nonrestrictive clause, isn't it?

Reply:
I see your point but it feels more like a restrictive clause to me (emphasis on the wrapping in a VectorArray)...